sympy/sets/fancysets.py

```diff
@@ -731,6 +755,9 @@ def is_finite_set(self):
         return self.size.is_finite
 
     def __bool__(self):
+        # this only distinguishes between definite null range
+        # and non-null/unknown null; getting True doesn't mean
+        # that it actually is not null
         return self.start != self.stop
```

Comment:
I think it would be better if this returned something like `is_eq(self.start, self.stop)` and raised an exception in the None case.